LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -1269,7 +1289,14 @@ extension DoseStore {
                     }
                     return dose.trimmed(to: basalDosingEnd)
                 }
-                let glucoseEffects = trimmedDoses.glucoseEffects(insulinModel: insulinModel, insulinSensitivity: insulinSensitivitySchedule)
+                // Update the longest effect duration based on the doses retrieved
+                // ANNA TODO: any way to make this more "clean"? Essensially, the aim is to find the longest effect duration among the doses
+                self.longestEffectDuration = trimmedDoses.max {
+                    ($0.insulinModel?.effectDuration ?? defaultInsulinModel.effectDuration)
+                    < ($1.insulinModel?.effectDuration ?? defaultInsulinModel.effectDuration)}?.insulinModel?.effectDuration
+                    ?? self.longestEffectDuration
```

Comment:
Is there any way to make this more "clean"? Essentially, this line is finding the longest effect duration among the doses, defaulting back on the current longestEffectDuration if the doses don't have any associated insulin models.

Reply:
You can clear out the `nil` values then grab the maximum:
```swift
trimmedDoses.compactMap({ $0.insulinModel?.effectDuration }).max() ?? self.longestEffectDuration
```